pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -317,6 +335,242 @@ func (b *BridgePodInterface) createBridge() error {
 	return nil
 }
 
+type MasqueradePodInterface struct {
+	vif                 *VIF
+	iface               *v1.Interface
+	podNicLink          netlink.Link
+	domain              *api.Domain
+	podInterfaceNum     int
+	podInterfaceName    string
+	bridgeInterfaceName string
+	vmNetworkCIDR       string
+	gatewayAddr         *netlink.Addr
+}
+
+func (p *MasqueradePodInterface) discoverPodNetworkInterface() error {
+	link, err := Handler.LinkByName(p.podInterfaceName)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", p.podInterfaceName)
+		return err
+	}
+	p.podNicLink = link
+
+	// Get interface MTU
+	p.vif.Mtu = uint16(p.podNicLink.Attrs().MTU)
```

Comment:
For me this conversion looks dangerous, the range of `uint16` is 0 to 65535 and MTU is `int` type. I would add additional checks.

Reply:
Okay, but what about `jumbograms`? Do we support it or not?
According to RFC [doc](https://tools.ietf.org/html/rfc2675)

> A "jumbogram" is an IPv6 packet containing a payload longer than
   65,535 octets.  This document describes the IPv6 Jumbo Payload
   option, which provides the means of specifying such large payload
   lengths.  It also describes the changes needed to TCP and UDP to make
   use of jumbograms.

> Jumbograms are relevant only to IPv6 nodes that may be attached to
   **links with a link MTU greater than 65,575 octets**, and need not be
   implemented or understood by IPv6 nodes that do not support
   attachment to links with such large MTUs.

So if some crazy guy will need to use it then this conversion will fail.
Or maybe there is some check somewhere else which prevents using such a high MTU.

Reply:
Ah forget about `jumbogram` comment, `virtio` does not support it. But I still think there should be some additional check or `VIF.Mtu` field should be the same type. It's just my opinion after all, it's up to you.

Reply:
I think it will be hard to chnage the variable to int because of the dhcp option
```go
	mtuArray := make([]byte, 2)
	binary.BigEndian.PutUint16(mtuArray, mtu)

	dhcpOptions := dhcp.Options{
		dhcp.OptionSubnetMask:       []byte(clientMask),
		dhcp.OptionRouter:           []byte(routerIP),
		dhcp.OptionDomainNameServer: bytes.Join(dnsIPs, nil),
		dhcp.OptionInterfaceMTU:     mtuArray,
	}
```

I can add if statement to check this < 0 and > 65535

Reply:
I think it makes sense to guard the code and not rely on virtio limitations. A if statement is the way to go.

Reply:
That being said... Isn't it the same bug that is perhaps present in other places of code? We may want to fix them too. I would be fine if e.g. this part is moved into a separate PR, as long as we are assured this happens.

Reply:
I see this bug in at least one other place, for BridgePodInterface. (Not a surprise considering that a lot of this code is a copy-pasted and then adjusted code for bridge implementation.) @SchSeba please tell us what you prefer: fixing the issue here or sending a follow up PR. I am fine with both approaches. (I am not fine with leaving it not handled for long though.)

Reply:
@booxter I just add this to the code for the bridge and the masquerade bindings. 
Also added a unit test for that please see my last commit